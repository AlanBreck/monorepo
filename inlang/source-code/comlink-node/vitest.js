// eslint-disable-next-line no-undef
export const WorkerPrototype = globalThis.Worker
export const adapter = (x) => x
