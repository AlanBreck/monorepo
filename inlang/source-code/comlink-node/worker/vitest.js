// @ts-ignore
export const endpoint = self
