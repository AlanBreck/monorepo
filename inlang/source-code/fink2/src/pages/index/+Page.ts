import { html } from "lit"
import "../../layout.js"
import { property, state } from "lit/decorators.js"
import { BaseElement } from "../../baseElement.js"
import type { PageContext } from "vike/types"

export const PageElementTag = "fink-index-page"

export class Page extends BaseElement {
	@property()
	pageContext!: PageContext

	@state()
	count = 0

	render() {
		return html`
			<fink-layout>
				<p>Welcome to subpage!</p>
				<button
					@click=${() => {
						this.count++
						console.log("count", this.count)
					}}
				>
					Increase Count: ${this.count}
				</button>
				<p>Current site is ${this.pageContext}</p>
				<a href="/subpage">Navigate to sub-page</a>
				<p class="text-red-500">testing tailwind</p>
			</fink-layout>
		`
	}
}
