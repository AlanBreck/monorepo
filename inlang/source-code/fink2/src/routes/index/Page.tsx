import { useEffect } from "react";
import Layout from "../../layout.tsx";
import { poll } from "../../poll.ts";
import { projectAtom, selectedProjectPathAtom } from "../../state.ts";
import { useAtom } from "jotai";
import { loadProjectFromOpfs } from "@inlang/sdk2";
// import { MessageBundle } from "../../components/InlangBundle.tsx";
// import { mockBundle } from "../../mock/mockHelper.ts";

export default function App() {
	const [project, setProject] = useAtom(projectAtom);
	const [selectedProjectPath] = useAtom(
		selectedProjectPathAtom
	);

	useEffect(() => {
		console.log("project on page", project)
		poll({
			every: 2000,
			fn: async () => {
				if (selectedProjectPath) {
					const project = await loadProjectFromOpfs({ path: selectedProjectPath });
					setProject(project);
				}
			},
		});
	}, []);

	console.log("App rerendered")
	return (
		<>
			<Layout>
				<p>{JSON.stringify(project?.settings.get()) ?? "fallback"}</p>
				{/* <MessageBundle
					// key={bundle.id}
					bundle={mockBundle}
					settings={project?.db.settings.get()}
					changeMessageBundle={onBundleChange as any}
					insertMessage={onMesageInsert as any}
					updateMessage={onMesageUpdate as any}
					insertVariant={onVariantInsert as any}
					updateVariant={onVariantUpdate as any}
					deleteVariant={onVariantDelete as any}
					filteredLocales={filteredLocales.length > 0 ? filteredLocales : undefined}
					fixLint={(e: any) => {
						const { fix, lintReport } = e.detail.argument as {
							fix: string
							lintReport: LintReport
						}
		
						project.fix(lintReport, { title: fix })
					}}
				/> */}
			</Layout>
		</>
	);
}
