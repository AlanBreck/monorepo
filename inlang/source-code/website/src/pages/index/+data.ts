import { privateEnv } from "@inlang/env-variables"

export { data }
export type Data = Awaited<ReturnType<typeof data>>

async function data() {
	try {
		// prettier-ignore
		const raw = JSON.stringify({
			"refresh": true,
			"query": {
				"kind": "InsightVizNode",
				"source": {
				"kind": "TrendsQuery",
				"properties": {
					"type": "AND",
					"values": [
					{
						"type": "AND",
						"values": []
					}
					]
				},
				"filterTestAccounts": true,
				"dateRange": {
					"date_to": null,
					"date_from": "-180d"
				},
				"series": [
					{
					"kind": "EventsNode",
					"event": "IDE-EXTENSION activated",
					"name": "IDE-EXTENSION activated",
					"custom_name": "Entire ecosystem",
					"math": "dau"
					},
					{
					"kind": "EventsNode",
					"event": "EDITOR clicked in field",
					"name": "EDITOR clicked in field",
					"math": "dau"
					},
					{
					"kind": "EventsNode",
					"event": "SDK loaded project",
					"name": "SDK loaded project",
					"properties": [
						{
						"key": "appId",
						"type": "event",
						"value": [
							"library.inlang.paraglideJs"
						],
						"operator": "exact"
						}
					],
					"math": "unique_group",
					"math_group_type_index": 1
					}
				],
				"interval": "month",
				"breakdownFilter": {
					"breakdown_group_type_index": 1
				},
				"trendsFilter": {
					"showLegend": false,
					"formula": "A + B + C",
					"display": "ActionsLineGraphCumulative",
					"showValuesOnSeries": true
				}
				},
				"full": true
			}
		})

		const response = await fetch(
			`https://eu.posthog.com/api/projects/${privateEnv.PUBLIC_POSTHOG_PROJECT_ID}/query`,
			{
				method: "POST",
				headers: {
					Authorization: "Bearer " + privateEnv.POSTHOG_API_KEY,
					"Content-Type": "application/json",
				},
				body: raw,
				redirect: "follow",
				cache: "no-cache",
			}
		)
		const json = await response.json()
		const number = json.results[0].data.at(-1)

		if (number) {
			return {
				projectCount: number,
			}
		} else {
			return {
				projectCount: "10000+",
			}
		}
	} catch (error) {
		return {
			projectCount: "10000+",
		}
	}
}
