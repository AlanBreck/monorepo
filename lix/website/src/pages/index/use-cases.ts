type UseCase = {
	title: string
	year: number
	link: string
}

export const useCases: Array<UseCase> = [
	{
		title: "Fink - Change Control in Translation Management Systems",
		year: 2024,
		link: "https://opral.substack.com/cp/146431448",
	},
]
