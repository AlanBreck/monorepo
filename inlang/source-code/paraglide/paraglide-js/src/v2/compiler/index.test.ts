import { compileMessage } from "./index.js"
import { describe, expect, test } from "vitest"
import type { Message } from "@inlang/sdk/v2"

describe("v2 - compileMessage", () => {
	test("no inputs, no selectors", () => {
		const message: Message = {
			locale: "en",
			declarations: [],
			selectors: [],
			variants: [
				{
					match: [],
					pattern: [
						{
							type: "text",
							value: "Hello world",
						},
					],
				},
			],
		}

		const compiled = compileMessage({ message, bundleId: "hello_world" })
		expect(compiled.source).toMatchSnapshot()
	})

	test("simple input, no selectors", () => {
		const message: Message = {
			locale: "en",
			declarations: [
				{
					type: "input",
					name: "name",
					value: {
						type: "expression",
						annotation: undefined,
						arg: { type: "variable", name: "name" },
					},
				},
			],
			selectors: [],
			variants: [
				{
					match: [],
					pattern: [
						{
							type: "text",
							value: "Hello ",
						},
						{
							type: "expression",
							annotation: undefined,
							arg: { type: "variable", name: "name" },
						},
					],
				},
			],
		}

		const compiled = compileMessage({ message, bundleId: "hello_name" })
		expect(compiled.source).toMatchSnapshot()
	})

	test("input with annotation, no selectors", () => {
		const message: Message = {
			locale: "en",
			declarations: [
				{
					type: "input",
					name: "name",
					value: {
						type: "expression",
						annotation: {
							type: "function",
							name: "plural",
							options: [],
						},
						arg: { type: "variable", name: "name" },
					},
				},
			],
			selectors: [],
			variants: [
				{
					match: [],
					pattern: [
						{
							type: "text",
							value: "Hello ",
						},
						{
							type: "expression",
							annotation: undefined,
							arg: { type: "variable", name: "name" },
						},
					],
				},
			],
		}

		const compiled = compileMessage({ message, bundleId: "hello_name" })
		expect(compiled.source).toMatchSnapshot()
	})

	test("input with annotation and selectors", () => {
		const message: Message = {
			locale: "en",
			declarations: [
				{
					type: "input",
					name: "name",
					value: {
						type: "expression",
						annotation: {
							type: "function",
							name: "plural",
							options: [],
						},
						arg: { type: "variable", name: "name" },
					},
				},
			],
			selectors: [
				{
					type: "expression",
					annotation: undefined,
					arg: {
						type: "variable",
						name: "name",
					},
				},
			],
			variants: [
				{
					match: ["*"],
					pattern: [
						{
							type: "text",
							value: "Hello ",
						},
						{
							type: "expression",
							annotation: undefined,
							arg: { type: "variable", name: "name" },
						},
					],
				},
			],
		}

		const compiled = compileMessage({ message, bundleId: "hello_name" })
		expect(compiled.source).toMatchSnapshot()
	})
})
