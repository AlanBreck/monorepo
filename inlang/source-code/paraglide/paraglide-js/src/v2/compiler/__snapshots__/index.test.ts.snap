// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`v2 - compileMessage > input with annotation and selectors 1`] = `
"export const hello_name = (inputs) => {
    input.name = registry.plural(inputs.name)
	const selector_0 = inputs.name

    if(true) return \`Hello \${inputs.name}\`
}"
`;

exports[`v2 - compileMessage > input with annotation, no selectors 1`] = `
"export const hello_name = (inputs) => {
    inputs.name = registry.plural(inputs.name)
    return \`Hello \${inputs.name}\`
}"
`;

exports[`v2 - compileMessage > no inputs, no selectors 1`] = `"export const hello_world = () => \`Hello world\`"`;

exports[`v2 - compileMessage > simple input, no selectors 1`] = `"export const hello_name = (inputs) => \`Hello \${inputs.name}\`"`;
